diff --git a/Library/Homebrew/extend/ENV/shared.rb b/Library/Homebrew/extend/ENV/shared.rb
index a4ed78a78..768921473 100644
--- a/Library/Homebrew/extend/ENV/shared.rb
+++ b/Library/Homebrew/extend/ENV/shared.rb
@@ -260,7 +260,7 @@ module SharedEnvExtension
     if @build_bottle && @bottle_arch
       @bottle_arch.to_sym
     else
-      Hardware.oldest_cpu
+      :native
     end
   end
 
diff --git a/Library/Homebrew/extend/ENV/super.rb b/Library/Homebrew/extend/ENV/super.rb
index ba3015a62..3a649a26a 100644
--- a/Library/Homebrew/extend/ENV/super.rb
+++ b/Library/Homebrew/extend/ENV/super.rb
@@ -66,6 +66,9 @@ module Superenv
     self["HOMEBREW_DEPENDENCIES"] = determine_dependencies
     self["HOMEBREW_FORMULA_PREFIX"] = @formula.prefix unless @formula.nil?
 
+    append CC_FLAG_VARS, "-march=native"
+    append "RUSTFLAGS", "-C target-cpu=native"
+
     # The HOMEBREW_CCCFG ENV variable is used by the ENV/cc tool to control
     # compiler flag stripping. It consists of a string of characters which act
     # as flags. Some of these flags are mutually exclusive.
diff --git a/Library/Homebrew/formula_installer.rb b/Library/Homebrew/formula_installer.rb
index 980ff92b9..74d13ce37 100644
--- a/Library/Homebrew/formula_installer.rb
+++ b/Library/Homebrew/formula_installer.rb
@@ -125,47 +125,13 @@ class FormulaInstaller
   end
 
   def pour_bottle?(install_bottle_options = { warn: false })
-    return false if @pour_failed
-
-    return false if !formula.bottled? && !formula.local_bottle_path
     return true  if force_bottle?
-    return false if build_from_source? || build_bottle? || interactive?
-    return false if cc
-    return false unless options.empty?
-    return false if formula.bottle_disabled?
-
-    unless formula.pour_bottle?
-      if install_bottle_options[:warn] && formula.pour_bottle_check_unsatisfied_reason
-        opoo <<~EOS
-          Building #{formula.full_name} from source:
-            #{formula.pour_bottle_check_unsatisfied_reason}
-        EOS
-      end
-      return false
-    end
-
-    bottle = formula.bottle_specification
-    unless bottle.compatible_cellar?
-      if install_bottle_options[:warn]
-        opoo <<~EOS
-          Building #{formula.full_name} from source:
-            The bottle needs a #{bottle.cellar} Cellar (yours is #{HOMEBREW_CELLAR}).
-        EOS
-      end
-      return false
-    end
-
-    true
+    false
   end
 
   def install_bottle_for?(dep, build)
     return pour_bottle? if dep == formula
-    return false if build_from_source_formulae.include?(dep.full_name)
-    return false unless dep.bottle && dep.pour_bottle?
-    return false unless build.used_options.empty?
-    return false unless dep.bottle.compatible_cellar?
-
-    true
+    false
   end
 
   def prelude
