diff --git a/Library/Homebrew/extend/ENV/shared.rb b/Library/Homebrew/extend/ENV/shared.rb
index 381ceccfd..baa92021e 100644
--- a/Library/Homebrew/extend/ENV/shared.rb
+++ b/Library/Homebrew/extend/ENV/shared.rb
@@ -275,7 +275,7 @@ module SharedEnvExtension
     if @build_bottle && @bottle_arch
       @bottle_arch.to_sym
     else
-      Hardware.oldest_cpu
+      :alderlake
     end
   end
 
diff --git a/Library/Homebrew/extend/ENV/super.rb b/Library/Homebrew/extend/ENV/super.rb
index c89a4604c..507a162dd 100644
--- a/Library/Homebrew/extend/ENV/super.rb
+++ b/Library/Homebrew/extend/ENV/super.rb
@@ -87,6 +87,9 @@ module Superenv
     self["HOMEBREW_DEPENDENCIES"] = determine_dependencies
     self["HOMEBREW_FORMULA_PREFIX"] = @formula.prefix unless @formula.nil?
 
+    append CC_FLAG_VARS, "-march=alderlake -mtune=alderlake -mno-avx512f"
+    append "RUSTFLAGS", "-C target-feature=-avx512 -C target-cpu=alderlake"
+
     # The HOMEBREW_CCCFG ENV variable is used by the ENV/cc tool to control
     # compiler flag stripping. It consists of a string of characters which act
     # as flags. Some of these flags are mutually exclusive.
diff --git a/Library/Homebrew/formula_installer.rb b/Library/Homebrew/formula_installer.rb
index 0a51b1fbb..6d16fcc16 100644
--- a/Library/Homebrew/formula_installer.rb
+++ b/Library/Homebrew/formula_installer.rb
@@ -144,50 +144,15 @@ class FormulaInstaller
   def pour_bottle?(output_warning: false)
     return false if !formula.bottle_tag? && !formula.local_bottle_path
     return true  if force_bottle?
-    return false if build_from_source? || build_bottle? || interactive?
-    return false if @cc
-    return false unless options.empty?
-    return false if formula.bottle_disabled?
-
-    unless formula.pour_bottle?
-      if output_warning && formula.pour_bottle_check_unsatisfied_reason
-        opoo <<~EOS
-          Building #{formula.full_name} from source:
-            #{formula.pour_bottle_check_unsatisfied_reason}
-        EOS
-      end
-      return false
-    end
-
-    return true if formula.local_bottle_path.present?
-
-    bottle = formula.bottle_for_tag(Utils::Bottles.tag.to_sym)
-    return false if bottle.nil?
-
-    unless bottle.compatible_locations?
-      if output_warning
-        prefix = Pathname(bottle.cellar).parent
-        opoo <<~EOS
-          Building #{formula.full_name} from source as the bottle needs:
-          - HOMEBREW_CELLAR: #{bottle.cellar} (yours is #{HOMEBREW_CELLAR})
-          - HOMEBREW_PREFIX: #{prefix} (yours is #{HOMEBREW_PREFIX})
-        EOS
-      end
-      return false
-    end
 
-    true
+    false
   end
 
   sig { params(dep: Formula, build: BuildOptions).returns(T::Boolean) }
   def install_bottle_for?(dep, build)
     return pour_bottle? if dep == formula
 
-    @build_from_source_formulae.exclude?(dep.full_name) &&
-      dep.bottle.present? &&
-      dep.pour_bottle? &&
-      build.used_options.empty? &&
-      dep.bottle&.compatible_locations?
+    false
   end
 
   sig { void }
diff --git a/Library/Homebrew/hardware.rb b/Library/Homebrew/hardware.rb
index 52bf2e5e8..50e8be790 100644
--- a/Library/Homebrew/hardware.rb
+++ b/Library/Homebrew/hardware.rb
@@ -28,6 +28,7 @@ module Hardware
       def optimization_flags
         @optimization_flags ||= {
           native:             arch_flag("native"),
+          alderlake:          "-march=alderlake",
           ivybridge:          "-march=ivybridge",
           sandybridge:        "-march=sandybridge",
           nehalem:            "-march=nehalem",
